File: modules/agent/src/schemas/opportunity.py
"""Opportunity schemas for API validation."""

from typing import List, Optional

from pydantic import BaseModel, Field


class OpportunityCreate(BaseModel):
    """Model for creating an opportunity."""
    account_type: str = "Organization"
    account: Optional[str] = None
    contacts: Optional[List[dict]] = None
    industry: Optional[List[str]] = None
    industry_ids: Optional[List[int]] = None
    title: str
    opportunity_name: str
    estimated_value: Optional[float] = None
    probability: Optional[int] = Field(default=None, ge=0, le=100)
    expected_close_date: Optional[str] = None
    description: Optional[str] = None
    status: str = "Qualifying"
    source: str = "manual"
    project_ids: Optional[List[int]] = None


class OpportunityUpdate(BaseModel):
    """Model for updating an opportunity."""
    account: Optional[str] = None
    contacts: Optional[List[dict]] = None
    title: Optional[str] = None
    opportunity_name: Optional[str] = None
    estimated_value: Optional[float] = None
    probability: Optional[int] = Field(default=None, ge=0, le=100)
    expected_close_date: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    source: Optional[str] = None
    project_ids: Optional[List[int]] = None

File: modules/agent/internal/schemas/opportunity.go
package schemas

// OpportunityCreate represents the request body for creating an opportunity
type OpportunityCreate struct {
	AccountType       string    `json:"account_type" validate:"omitempty,oneof=Organization Individual"`
	Account           *string   `json:"account"`
	Contacts          []Contact `json:"contacts"`
	Industry          []string  `json:"industry"`
	IndustryIDs       []int64   `json:"industry_ids"`
	Title             string    `json:"title" validate:"required"`
	OpportunityName   string    `json:"opportunity_name" validate:"required"`
	EstimatedValue    *float64  `json:"estimated_value" validate:"omitempty,min=0"`
	Probability       *int      `json:"probability" validate:"omitempty,min=0,max=100"`
	ExpectedCloseDate *string   `json:"expected_close_date"`
	Description       *string   `json:"description"`
	Status            string    `json:"status"`
	Source            string    `json:"source" validate:"omitempty,oneof=manual inbound referral event campaign agent"`
	ProjectIDs        []int64   `json:"project_ids"`
}

// OpportunityUpdate represents the request body for updating an opportunity
type OpportunityUpdate struct {
	Account           *string   `json:"account"`
	Contacts          []Contact `json:"contacts"`
	Title             *string   `json:"title"`
	OpportunityName   *string   `json:"opportunity_name"`
	EstimatedValue    *float64  `json:"estimated_value" validate:"omitempty,min=0"`
	Probability       *int      `json:"probability" validate:"omitempty,min=0,max=100"`
	ExpectedCloseDate *string   `json:"expected_close_date"`
	Description       *string   `json:"description"`
	Status            *string   `json:"status"`
	Source            *string   `json:"source"`
	ProjectIDs        []int64   `json:"project_ids"`
}
